### ISSUE TITLE
1 bouton open / close

### ISSUE THREAD

**Issue by hellosct1** [hellosct1/firefox-send#1]
Une demande pour un seul bouton pour ouvrir et fermer la barre verticale

**Comment by Skallwar** [comment]
Hi,

Why not make your icon toggle the sidebar ? I think it will be better than a sub menu

**Comment by Brodie-Hamilton** [comment]
I have added a PR to address this issue. Unfortunately due to limitations of web extensions making the button toggle the sidebar open AND closed is not trivially easy / has bugs. As such the `browser_action` button only opens/displays the sidebar and the user must close the sidebar itself to hide it.

**Comment by Skallwar** [comment]
Seems to be a good solution. Clicking on the icon to close the sidebar is not really important for me

**Comment by Skallwar** [comment]
Maybe using a static variables will do the trick? If isOpen - > close, else - > Open

**Comment by Brodie-Hamilton** [comment]
> Maybe using a static variables will do the trick? If isOpen - > close, else - > Open

I attempted to do this however the issue lies in the fact that there is no way to hook into when the user manually closes the sidebar (outside of clicking the icon) to keep the local variable in sync with the actual browser state. This often resulted in the user having to double click the button to trigger the opening (which I found more annoying than just not having the ability to toggle via the button itself).

**Comment by Skallwar** [comment]
This page suggest that it's possible to retrieve url form sidebar : https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/sidebarAction/getPanel
So if url = fierfoxsend -> close, else -> open should be a viable solution

**Comment by Brodie-Hamilton** [comment]
I don't think this solution is quite what we are after. I believe the `browser.sidebarAction.getPanel({})` method will return the global panel in the scope of the extension itself. If you were to set multiple panels within your extension you could use that method to determine which one was currently visible (in the context of your extension) but it cannot be used to determine which panel the user is actually seeing (if any).

Also unfortunately the `getPanel` method returns an asynchronous `Promise` which cannot be used to open or close the sidebar. Firefox requires the `open()` and `close()` methods to be [executed synchronously from a user input handler](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/User_actions).

**Comment by Skallwar** [comment]
It's a shame that it not possible, even if it's not really a big matter for the end user

@hellosct1 don't seem to watch his repo ^^
